compose/project.py

```diff
@@ -695,7 +695,9 @@ def push(self, service_names=None, ignore_push_failures=False):
             repo, tag, sep = parse_repository_tag(service.image_name)
             service_image_name = sep.join((repo, tag)) if tag else sep.join((repo, 'latest'))
 
-            if service_image_name not in unique_images:
+            # Add service to unique if it's pushable
+            if (service_image_name not in unique_images and
+                    service.can_be_pushed()):
                 service.push(ignore_push_failures)
```

Comment:
Unrelated: actually never given much attention to `ignore_push_failures`, but I think `ignore_push_failures` is way too broad; it seems to ignore _any_ push failure (for whatever cause push failed). I think it'd make sense to either (by default?) ignore  _missing_ images (e.g., if I didn't build a service yet and the image isn't present), or add / rename this option to `--ignore-missing`.

I don't think other failures should be ignorable (e.g. authentication failed, registry unreachable, or pushing caused any other failure)

@ulyssessouza WDYT?

Reply:
Opened https://github.com/docker/compose/issues/7431 to propose adding an `--ignore-missing` or `--skip-missing` flag